common/src/main/kotlin/com/hartwig/actin/molecular/evidence/actionability/IndirectEvidenceMatcher.kt

```diff
@@ -0,0 +1,107 @@
+package com.hartwig.actin.molecular.evidence.actionability
+
+import com.hartwig.actin.datamodel.molecular.driver.ProteinEffect
+import com.hartwig.actin.datamodel.molecular.driver.Variant
+import com.hartwig.actin.molecular.evidence.ServeVerifier.isCombinedProfile
+import com.hartwig.actin.molecular.evidence.curation.ApplicabilityFiltering
+import com.hartwig.actin.molecular.evidence.curation.GenericInhibitorFiltering
+import com.hartwig.actin.molecular.interpretation.GeneAlterationFactory
+import com.hartwig.serve.datamodel.Knowledgebase
+import com.hartwig.serve.datamodel.ServeRecord
+import com.hartwig.serve.datamodel.efficacy.EfficacyEvidence
+import com.hartwig.serve.datamodel.molecular.hotspot.KnownHotspot
+import com.hartwig.serve.datamodel.molecular.hotspot.VariantAnnotation
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+
+data class TreatmentKey(
+    val gene: String,
+    val proteinEffect: ProteinEffect
+)
+
+private data class HotspotKey(
+    val gene: String,
+    val chromosome: String,
+    val position: Int,
+    val ref: String,
+    val alt: String
+)
+
+class IndirectEvidenceMatcher(
+    private val treatmentByGeneEffectForHotspots: Map<TreatmentKey, Set<EfficacyEvidence>>
+) {
+
+    // TODO: Actually this returns indirect but also direct hits, caller needs to filter out direct hits.
+    //  better name?
+    fun findIndirectEvidence(variant: Variant): Set<EfficacyEvidence> {
+        return findIndirectEvidence(variant.gene, variant.proteinEffect)
+    }
+
+    private fun findIndirectEvidence(gene: String, proteinEffect: ProteinEffect): Set<EfficacyEvidence> {
+        return treatmentByGeneEffectForHotspots[TreatmentKey(gene, proteinEffect)] ?: emptySet()
+    }
+
+    companion object {
+
+        val logger: Logger = LogManager.getLogger(IndirectEvidenceMatcher::class.java)
+
+        fun create(serveRecord: ServeRecord): IndirectEvidenceMatcher {
+            val knownHotspotsByPosition = mapKnownHotspotsByPosition(serveRecord)
+
+            val treatmentByGeneEffect = serveRecord.evidences()
+                .asSequence()
+                .filter { it.source() == Knowledgebase.CKB }
+                .filterNot { isCombinedProfile(it.molecularCriterium()) }
+                .filter { it.molecularCriterium().hotspots().any() }
+                .mapNotNull { evidence -> matchableEvidence(evidence, knownHotspotsByPosition) }
+                .groupBy({ it.first }, { it.second })
+                .mapValues { (_, evidences) -> evidences.toCollection(HashSet()) }
+
+            return IndirectEvidenceMatcher(treatmentByGeneEffect)
+        }
+
+        fun empty(): IndirectEvidenceMatcher = IndirectEvidenceMatcher(emptyMap())
+
+        private fun mapKnownHotspotsByPosition(serveRecord: ServeRecord): Map<HotspotKey, KnownHotspot> {
```

Comment:
Maybe rename for clarity since you only extract hotspots associated with resistance? e.g. `mapKnownResistantHotspotsByPosition`?